plonky2/src/hash/merkle_tree.rs

```diff
@@ -90,6 +90,20 @@ fn fill_digests_buf<F: RichField, H: Hasher<F>>(
     leaves: &[Vec<F>],
     cap_height: usize,
 ) {
+    // Special case of a tree that's all cap. The usual case will panic because we'll try to split
+    // an empty slice into chunks of `0`. (We would not need this if there was a way to split into
+    // `blah` chunks as opposed to chunks _of_ `blah`.)
+    if digests_buf.is_empty() {
+        debug_assert_eq!(cap_buf.len(), leaves.len());
+        cap_buf
+            .par_iter_mut()
+            .zip(leaves)
+            .for_each(|(cap_buf, leaf)| {
+                cap_buf.write(H::hash(leaf, false));
+            });
+        return;
+    }
```

Comment:
A special case with an early return was the neatest way I found to do this. 

The closure in the normal case is correct when `subtree_digests.is_empty()` but Rust will not produce such a value. In other words, the _real_ bug here is Rust's well-meaning sanity checks…